src/qml/views/timeline/timeline.qml

```diff
@@ -232,6 +235,103 @@ Rectangle {
                                 currentTrack = index
                                 timeline.selectTrackHead(currentTrack)
                             }
+                            MouseArea {
+                                id: dragMouseArea
+                                anchors.top: parent.top
+                                anchors.bottom: parent.bottom
+                                anchors.left: parent.left
+                                hoverEnabled: true
+                                cursorShape: Qt.DragMoveCursor
+                                drag.target: dragItem
+                                drag.axis: Drag.YAxis
+                                drag.maximumY: trackHeaders.height - y - height / 2
+                                drag.minimumY : -y - height / 2
+                                width: containsMouse ? trackHead.width : 8
+                                Behavior on width { PropertyAnimation {easing.type: Easing.InOutCubic} }
+                                onReleased: {
+                                    if (drag.active) {
+                                        dragItem.Drag.drop()
+                                    }
+                                }
+                                Rectangle {
+                                    id: dragItem
+                                    property var trackHead: trackHead
+                                    color: activePalette.dark
+                                    opacity: 0.9
+                                    Drag.active: dragMouseArea.drag.active
+                                    Drag.hotSpot.x: width / 2
+                                    Drag.hotSpot.y: height / 2
+                                    Drag.keys: ["trackHeader"]
+                                    Label {
+                                        anchors.fill: parent
+                                        horizontalAlignment: Text.AlignHCenter
+                                        verticalAlignment: Text.AlignVCenter
+                                        elide: Text.ElideRight
+                                        text: qsTr("Move track %1").arg(trackHead.trackName)
+                                        color: activePalette.light
```

Comment:
I think the white on gray looks a little funny in the lighter themes, and maybe it should just follow the theme text color. The Overwrite/Insert block when dragging to the timeline uses black with a white outline, but I think that would look bad on this smaller size text.

Reply:
Thank you for mentioning the Overwrite/Insert block. I changed the styling to match that - including the larger text size. I think it is much better. Let me know what you think.

Also, I changed the narrow handle to be transparent until the mouse hovers over it. So not it is less obvious that there is a reason to put the mouse there, but the color is not visible all the time - only when hovering/dragging. I think it restores the original visual appeal.

Additionally, I also set the current track to whichever track is being hovered over (as the overwrite drag does). So the white panel is always over a consistent color background.

Have a look and see what you think.

Reply:
I think it might be a little better cosmetically except the only hint that this is draggable is the mouse pointer, which is subtle. I think a "Move %1" would be better (shorter than "Move track %1"). I was not asking to use the larger text size, which reduces the space for the text. Hmm... in fact, this rectangle width does not need to be limited to the header:
```
--- src/qml/views/timeline/timeline.qml
+++ src/qml/views/timeline/timeline.qml
@@ -248,3 +248,3 @@ Rectangle {
                                 drag.minimumY : -y - height / 2
-                                width: containsMouse ? trackHead.width : 8
+                                width: containsMouse ? root.width : 8
                                 Behavior on width { PropertyAnimation {easing.type: Easing.InOutCubic} }
@@ -267,3 +267,3 @@ Rectangle {
                                         anchors.fill: parent
-                                        text: trackHead.trackName
+                                        text: qsTr("Move %1").arg(trackHead.trackName)
                                         style: Text.Outline
```

But there is some clipping getting in the way of that until you drag, which might not be so bad. Maybe it is rather heavy to paint over the whole track when you accidentally hover into the track drag handle area.
![Screenshot from 2022-02-21 11-17-05](https://user-images.githubusercontent.com/1146683/155015253-21fca2a1-f528-4178-991f-25920c621bb0.png)

Reply:
For some reason, my change breaks the action.

Reply:
Here is a suggestion that sets the width to the text width and sets the width before drag starts:

```
--- a/src/qml/views/timeline/timeline.qml
+++ b/src/qml/views/timeline/timeline.qml
@@ -246,7 +246,7 @@ Rectangle {
                                 drag.axis: Drag.YAxis
                                 drag.maximumY: trackHeaders.height - y - height / 2
                                 drag.minimumY : -y - height / 2
-                                width: containsMouse ? trackHead.width : 8
+                                width: containsMouse | drag.active ? Math.max(parent.width, dragItemText.contentWidth + 20) : 8
                                 Behavior on width { PropertyAnimation {easing.type: Easing.InOutCubic} }
                                 onReleased: {
                                     if (drag.active) {
@@ -256,16 +256,20 @@ Rectangle {
                                 Rectangle {
                                     id: dragItem
                                     property var trackHead: trackHead
+                                    anchors.left: dragMouseArea.left
+                                    anchors.top: dragMouseArea.top
+                                    anchors.bottom: dragMouseArea.bottom
+                                    width: dragMouseArea.width
                                     color: 'white'
                                     Drag.active: dragMouseArea.drag.active
-                                    Drag.hotSpot.x: width / 2
                                     Drag.hotSpot.y: height / 2
                                     Drag.keys: ["trackHeader"]
                                     opacity: dragMouseArea.containsMouse | dragMouseArea.drag.active ? 0.7 : 0
                                     Behavior on opacity { NumberAnimation {} }
                                     Text {
+                                        id: dragItemText
                                         anchors.fill: parent
-                                        text: trackHead.trackName
+                                        text: qsTr("Move %1").arg(trackHead.trackName)
                                         style: Text.Outline
                                         styleColor: 'white'
                                         font.pixelSize: Math.min(Math.max(parent.height * 0.7, 15), 30)
@@ -274,14 +278,12 @@ Rectangle {
                                     }
                                     states: [
                                         State {
-                                            when: dragMouseArea.drag.active
+                                            when: dragMouseArea.drag.active | dragMouseArea.containsMouse
                                             ParentChange { target: dragItem; parent: trackHeaders.parent }
-                                            PropertyChanges { target: dragItem; anchors.fill: undefined }
                                         },
                                         State {
-                                            when: !dragMouseArea.drag.active
+                                            when: !dragMouseArea.drag.active & !dragMouseArea.containsMouse
                                             ParentChange { target: dragItem; parent: trackHead }
-                                            PropertyChanges { target: dragItem; anchors.fill: dragMouseArea }
                                         }
                                     ]
                                 }
```

![image](https://user-images.githubusercontent.com/821968/155039434-a9b34808-8ef2-455f-99c7-fed6593c3bc4.png)

Reply:
That looks good to me. If you like it too, go ahead and commit it.